src/praisonai-agents/praisonaiagents/tools/decorator.py

```
@@ -222,13 +252,23 @@ def decorator(fn: Callable) -> FunctionTool:
             logging.warning(f"Tool validation warning for {tool_instance.name}: {e}")
         
         # Register with global registry if available
+        # Note: Don't pass dynamic_schema_overrides again since FunctionTool already handles it
         try:
-            from .registry import get_registry
+            logging.debug(f"Attempting to import registry for tool {tool_instance.name}")
+            from praisonaiagents.tools.registry import get_registry
+            logging.debug(f"Successfully imported get_registry")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)